### PR TITLE
AEROGEAR-2175 basic module created and first check

### PR DESCRIPTION
**Pull request by StephenCoady**
### Motivation

[JIRA 1](https://issues.jboss.org/browse/AGDROID-2175) - Creating the base API
[JIRA 2](https://issues.jboss.org/browse/AGDROID-2127) - Implementing the first check

### Description

This is an initial to PR to verify that the approach is correct and that the protocols are ok. It has included one security check to show the flow of the code.

### CODE REVIEW

File: AGSSec.podspec [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Pod::Spec.new do |s|
+  s.name         = 'AGSSec'
+  s.version      = '1.0.0'
```
Comment (wtrocki) [inline AGSSec.podspec]: [Minor] Change version to follow others.

File: AGSSec.podspec [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  s.summary      = 'AeroGear Services security'
```
Comment (wtrocki) [inline AGSSec.podspec]: `OWASP security checks`

File: SecurityCeck.swift [hunk 3]
```diff
@@ -0,0 +1,16 @@
+//
+//  SecurityCeck.swift
+//  AGSSec
+//
+//  Created by Stephen Coady on 10/05/2018.
```
Comment (wtrocki) [inline SecurityCeck.swift]: Configure xcode to not include those:

File: SecurityCheckResult.swift [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+//  AGSSec
+//
+//  Created by Stephen Coady on 10/05/2018.
+//
+
+import Foundation
+
+public struct SecurityCheckResult {
+    
+    var name: String;
```
Comment (wtrocki) [inline SecurityCheckResult.swift]: Replace with `var` => `let` to make this immutable.

File: checks/IsEmulatorCheck.swift [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import Foundation
+
+open class IsEmulatorCheck: SecurityCheck{
+    var name = "Is Emulator Check";
+
+    internal func check() -> SecurityCheckResult {
+        var result = SecurityCheckResult.init(name: self.name, passed: true);
+        #if (arch(i386) || arch(x86_64) && os(iOS))
+        result.passed = false;
+        return result;
```
Comment (wtrocki) [inline checks/IsEmulatorCheck.swift]: Return should be moved out of the directive.
This is kinda funny bug that will crash app on normal phone.

File: modules/security/AgsSec.swift [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+/**
+ AeroGear Services Security SDK
+ 
+ SDK provides a way for the developer to perform security
+ checks on the device their code is running on
+ */
+open class AgsSec {
```
Comment (wtrocki) [inline modules/security/AgsSec.swift]: replace `open` => `public`

File: modules/security/AgsSec.swift [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+ 
+    func check(check: SecurityCheck) -> SecurityCheckResult {
```
Comment (wtrocki) [inline modules/security/AgsSec.swift]: ` check(check: SecurityCheck)` =>  check(_ check: SecurityCheck)

**Comment by wtrocki** [comment]
@StephenCoady World class work!

**Comment by jhellar** [comment]
There is typo in `SecurityCeck.swift`. `SecurityCeck.swift`, `SecurityCheckResult.swift` files and `checks` folder should be moved to `modules/security/`.

File: modules/security/SecurityCheckResult.swift [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import Foundation
+
+public struct SecurityCheckResult {
+    
+    let name: String;
+    var passed: Bool;
```
Comment (wtrocki) [inline modules/security/SecurityCheckResult.swift]: let it be here as well :D

**Comment by wtrocki** [comment]
@StephenCoady Looks like this is no longer WIP. There is no way to test it at the moment so let's add some comments/tests and merge it :)

**Comment by StephenCoady** [comment]
Thanks @wtrocki and @jhellar. Made all changes and I think it is ready to go now if you have time to review again.

**Comment by wtrocki** [comment]
@StephenCoady Amazing job!  I would love to see more documentation but that could be done in follow up work

**Comment by StephenCoady** [comment]
@wtrocki we have a separate ticket to do the docs so after we integrate all the tests that will be the first job.

**Pull request merged**